starling/filters/FragmentFilter.hx

```diff
@@ -321,37 +322,33 @@ class FragmentFilter
         context.setVertexBufferAt(mVertexPosAtID, null);
         context.setVertexBufferAt(mTexCoordsAtID, null);
         context.setTextureAt(mBaseTextureID, null);
+        Starling.current.removeCurrentProgram(); 
 
         support.popMatrix();
         support.popMatrix3D();
-
+        
         if (intoCache)
         {
             // restore support settings
             support.projectionMatrix.copyFrom(projMatrix);
-            support.projectionMatrix3D.copyFrom(projMatrix3D);
+            support.projectionMatrix3D = projMatrix3D;
             support.renderTarget = previousRenderTarget;
             support.popClipRect();
             
             // Create an image containing the cache. To have a display object that contains
-            // the filter output in object coordinates, we wrap it in a QuadBatch: that way,
-            // we can modify it with a transformation matrix.
+            // the filter output in object coordinates. That way we can modify it with a transformation matrix.
             
-            var quadBatch:QuadBatch = new QuadBatch();
-            var image:Image = new Image(cacheTexture);
+            var image: Image = new Image(cacheTexture);
```

Comment:
Not sure if type declarations in Starling are consistent across the while codebase, but at least in this file it seems to be most often without the space.